migrations/20170428154714_multiple_api_tokens/up.sql

```diff
@@ -0,0 +1,16 @@
+CREATE TABLE api_tokens (
+  id SERIAL PRIMARY KEY,
+  user_id integer NOT NULL REFERENCES users(id),
+  token character varying DEFAULT random_string(32) NOT NULL UNIQUE,
```

Comment:
The random_string function is using a 62 character dictionary, giving a 32 character string [~190 bits of randomness](https://www.wolframalpha.com/input/?i=log2(62%5E32)). That makes it more collision resistant than a UUID, and UUIDs are already collision resistant enough that you're recommended not to ever bother checking for a collision.

Using an approximation to the [birthday problem](https://en.wikipedia.org/wiki/Birthday_problem) the chance of having two colliding tokens in the first 10,000 users is [~2.2×10<sup>-50</sup>](https://www.wolframalpha.com/input/?i=1+-+e%5E((-(10000%5E2))+%2F+(2+*+62+%5E+32))), if every person on Earth were to create two accounts the chance of any two colliding is [~5×10<sup>-38</sup>](https://www.wolframalpha.com/input/?i=1+-+e%5E((-(15000000000%5E2))+%2F+(2+*+62+%5E+32))).

But yeah, if `random_string` somehow does create a duplicate it would raise a constraint violation, which would presumably be returned as an error in [`token::new`](https://github.com/rust-lang/crates.io/pull/697/files#diff-8eaf0fafad7d2ed9ec9a701c04424c96R75), which I assume `conduit` or some sort of middleware would transform to a 500 error response. Once I re-add the client-side error handling that should then be shown to user somehow, even if it's just a generic please try again message.